freqtrade/plot/plotting.py

```diff
@@ -267,7 +270,7 @@ def generate_profit_graph(pairs: str, tickers: Dict[str, pd.DataFrame],
     df_comb = create_cum_profit(df_comb, trades, 'cum_profit')
 
     # Plot the pairs average close prices, and total profit growth
-    avgclose = go.Scattergl(
+    avgclose = go.Scatter(
```

Comment:
I am not certain about this change (and the one in `add_profit()`)...

Did this usage cause some problems or crashes for you?

It's obviously problematic for Bollingerbands due to the very specific rendering (as a Band) ... but works fine for other uses on my machine.

The reason to use `Scattergl` over `Scatter`is performance for larger plots, since `Scatter` creates them as SVG objects (which causes rendering problems in the browser with a certain amount of points), while GL uses a more optimized method of rendering, allowing a LOT more points in one plot.

Reply:
I removed all occurrences of Scattergl out of an abundance of caution. My assumption is that webgl and plotly may still have some problems upstream and if a chart element is resource intensive enough to benefit from webgl, then it will probably break. If it does not break then it is probably not resource intensive and won't suffer from rendering performance problems.

You might consider waiting to implement webgl until plotly proves that it is not buggy. 

I can revert these changes and only change the BB to Scatter if you like. It's up to you.

Reply:
the problem with BB rendering is not resource intensiveness, but a simple programming error (with python short-circuiting breaking something too early...).

personally i would leave `Scattergl` for all other operations until we face issues ... especially for longer-term Profit-plots (multiple months) the speed-improvements are non neglectable

Reply:
I have noticed that volume does not display once a timerange crosses some threshold. For example 1 month of 5min tickers does not show the volume bars. Have you seen that behavior and is it possibly related? Do you want me to log an issue?

Reply:
I experimented with rendering using `Scattergl` and found that any inclusions broke something. When I reverted the substitutions I made in the last commit and then rendered bollinger bands without gl. That broke the profit graph. Wherever I used `Scattergl`, the chart displayed incorrectly.

I tried restarting my jupyter server and restarting the kernel after making code changes and calling `%autoreload`. This occurred in Brave, Chrome and Safari so I'm confident that this is a real bug with `Scattergl` and not just bollinger bands.

I'm using a Macbook air so perhaps that is resource constrained but I can't test on my other machine until my new CPU cooler arrives. Still, I wouldn't expect two weeks of 5min tickers to cause any problems.

You may want to confirm on a different machine or consider merging this PR without `Scattergl`.

Reply:
when testing such things - i would reccomend to test them as mainly intendet - using the cli interface, which (by sharing the cli command used) should allow us to recreate the issue you're seeing).
Even using ! runs it "slightly" different and in a jupyter context, which could have unknown consequences and can taint results.

I've just tried both `plot-profit` and `plot-dataframe` with 1.5 months 5m plots (using develop) and found no issue (other than bollinger not rendering correctly).

Also volume is there (but barely visible, since the bars tiny, and only visible once you zoom into the picture to only see a day or two).

Please also verify the version of plotly - it should be 4.1.1 at the moment (i'm lazy at updating this too sometimes ...)

Reply:
plotly==4.1.1

I tracked the bug to `fig.show(renderer='browser')` from a notebook.
With the exception of the BB, plotting works from the cli and the charts display correctly inline in a notebook. However,  `scattergl` does not work correctly when jupyter opens plots in the browser. 

I also applied some stying to the volume bars to make them more visible.

Reply:
so the scattergl problem is a jupyter problem, not a problem happening when using the cli?

(now i'm not against this change if it creates problems, it's more to understand the problem).

Reply:
Correct. In 83e596c I reverted `avgclose =` and `profit =` back to `go.Scattergl`.

When using Scatter.gl:

|                   |        cli           | jupyter inline | jupyter `renderer='browser'`|
| ---------- | ------------ | ------------- | ---------------------------- |
| BB | X | X | X |
| `avgclose = go.Scatter.gl` |           + | + | X |
| `profit = go.Scatter.gl` | + | + | X |

\+ = works
X = broken

Reply:
this finding about scattergl not working in browser renderer might be worth reporting to plotly so they can fix it in a future release.

However, it's fine to switch everything to `Scatter` for now.
Please have  a look at the tests. The failures are logical considering the changes done.
Any issues with that let me know and i'll lend you a hand with them.